File: reserve.pl
#!/usr/bin/env perl
# refer to http://forums.whirlpool.net.au/forum-replies.cfm?t=2313467&p=98#r1953

use strict;
use warnings;
use JSON qw( decode_json );
use LWP::Simple;
use Email::Send;
use Email::Send::Gmail;
use Email::Simple::Creator;

#my $stores = get("https://reserve.cdn-apple.com/AU/en_AU/reserve/iPhone/stores.json");
my $availability = get("https://reserve.cdn-apple.com/AU/en_AU/reserve/iPhone/availability.json");

my $brisbane = decode_json($availability)->{'R466'}->{'MGAK2X/A'}; # + 64 gold

if ($brisbane) {
my $email = Email::Simple->create(
header => [
From => 'YOUR_GMAIL_ADDRESS@gmail.com',
To => 'YOUR@EMAIL.com',
Subject => 'iPhone+',
],
body => 'Now available at Brisbane store.',
);

my $sender = Email::Send->new(
{ mailer => 'Gmail',
mailer_args => [
username => 'YOUR_GMAIL_ADDRESS@gmail.com',
password => 'YOUR_GMAIL_PASSWORD',
]
}
);
eval { $sender->send($email) };
die "Error sending email: $@" if $@;
}

File: reserve.js
//refer http://runnable.com/VCkxZdaHSM4nSf-X/check-iphone-6-stock-for-node-js
//node reserve.js

var https = require('https');

// models at http://www.thebookyard.com/?cPath=256_251_231_352
// stores at https://reserve.cdn-apple.com/GB/en_GB/reserve/iPhone/stores.json
var url = "https://reserve.cdn-apple.com/GB/en_GB/reserve/iPhone/availability.json";
var model = "MG4F2B/A";
var store = "R269";

function checkAvailability() {
  console.log("Checking Apple for stock...");
  
  https.get(url, function(res) {
    var body = '';

    res.on('data', function(chunk) {
        body += chunk;
    });

    res.on('end', function() {
        var json = JSON.parse(body);
        //console.log("Got response: ", json);
        console.log(model + " in stock? " + json[store][model]);
        console.log("Please wait...");
    });
  }).on('error', function(e) {
        console.log("Got error: ", e);
  });
  
  setTimeout(checkAvailability, 10000);
}

checkAvailability();
